ledokku-bootstrap.sh

```diff
@@ -1,52 +1,174 @@
+  if [ "$(whoami)" == "root" ] ; then
+      echo "${YELLOW}Nice you are running the script as root!${END}"
+  else
+      echo "${RED}Please! Run the script with root access${END}, without root access I cannot create plugins in Dokku"
+      exit
+  fi
+}
+
+function system-update {
+
+  # Updating and Upgrading system
+  ## If dokku was installed, it will be automatically updated to the latest version
+  ## Staying up to date is always good
+  if (whiptail --title "Update and Upgrade System " --yes-button "Yes" --no-button "Skip"  --yesno "Do you wish to Update packges and Upgrade your system?" 10 60) then
+      echo "You chose to update your system"
+      # Update and skip to next step
+      echo "${YELLOW}Updating System${END}"
+      sudo dpkg --configure -a
+      sudo apt -y --purge autoremove
+      sudo apt install -f
+      sudo apt -y update
+      wait
+      echo "${YELLOW}Upgrading System${END}"
+      sudo apt -y upgrade
+      sudo apt -y autoclean
+      sudo apt -y --purge autoremove &
+      process_id=$!
+      wait $process_id
+      echo "Exit status: $?"
+  else
+      echo "${YELLOW}You chose to skip.${END}"
+      # Should skip to next step
+  fi
+}
+
+function ensure-dokku() {
+  
+  # Confirming the existance of Dokku
+  ## If exists, promt for update
+  ## If not exits Let him download.
+  if which dokku >/dev/null; then
+      echo "${GREEN}Dokku Exists${END}"
+      # Promt for update
+      if (whiptail --title "Updating Dokku" --yes-button "Update" --no-button "Skip"  --yesno "Would you like to update your Dokku?" 10 60) then
+          echo "${YELLOW}You chose Update.${END}"
+          # Update Dokku
+          echo "${YELLOW}Upgrading Dokku${END}"
+          sudo apt-get -y update -qq
+          wait
+          sudo apt-get -qq -y --no-install-recommends install dokku herokuish sshcommand plugn gliderlabs-sigil dokku-update dokku-event-listener
+          wait
+          sudo apt -y upgrade &
+          process_id=$!
+          wait $process_id
+          echo "Exit status: $?"
+          echo "${YELLOW}Now you have the latest Version of Dokku${END}"
+          # Dokku Updated
+      else
+          echo "${Yellow}You chose to skip dokku updates.${END}"
+          # Dokku Update skipped
+      fi
+  else
+      echo "${RED}Dokku does not exist${END}"
+      # Show messagebox and make it mandatory to download and install dokku
+      whiptail --title "Unable to Detect Dokku" --msgbox "If you want to insatll your app using ledokku, it is madatory to install Dokku. So, I would like to install Dokku on behalf of you." 10 60
+      wait
+      echo "${YELLOW}Downloading Dokku from its Official Repository${END}"
+      wget https://raw.githubusercontent.com/dokku/dokku/v0.24.10/bootstrap.sh
+      wait
+      sudo DOKKU_TAG=v0.24.10 bash bootstrap.sh &
```

Comment:
We can actually make this `DOKKU_TAG` to have defaulted at least version and also give the users an option to change it to a specified version.

This specified version can be an `input from user` or `choose among the supported version, we can offer`

Reply:
To limit the number of interactions a user has to do let's do it the same way as we do for `LEDOKKU_TAG`, if the user sets it before the installation script is run we use the one provided by the user, otherwise we default to v2.24.10

Reply:
As we removed the upgrade option, let's keep it to the latest version and remove user interaction. 

Rather than creating this DOKKU_TAG as script variables, let's make DOKKU_TAG a global variable in the entire repository. This way, if the user will have the ability to change it, in case if he prefers.

Reply:
Just saw you removed the upgrade option, but what do you think about this instead? https://github.com/ledokku/ledokku/pull/393#discussion_r676130185